### ISSUE TITLE
v6 hiding tabbar with {display:none} does not work on iOS 

### ISSUE THREAD

**Issue by tarikpnr** [software-mansion/react-native-screens#1366]
### Current behavior

I am trying to achieve dynamically hiding tab bar behavior. I already know [what documentation suggest][1] for this purpose.

However, I am not a big fan of restructuring whole navigators and routes for only hiding tab bar. Thus, I did prefer a more dynamic way. But, I am facing a problem with IOS simulator/real device. It works fine on android devices.

What I do is basically, (just a summary of the logic I do app

```  
  <BottomTabBar
    style={{ display: shouldRenderTabBar ? "flex" : "none" }}
    {...props}
  />;

```

https://user-images.githubusercontent.com/61876765/158449077-0ebc3efc-e87a-4356-918a-418944621712.mov

And this is what happens on ios simulator/real-device. [Video link][2]

Seems like `display:none` doesn't affect anyway to height of bottom tab bar. I am wondering while it works on android emulator/real-device what's wrong with IOS? 
  

  [1]: https://reactnavigation.org/docs/hiding-tabbar-in-screens/
  [2]: https://streamable.com/n5xqej

### Expected behavior

Hide bottom tab bar with the applied styling

### Reproduction

https://snack.expo.dev/@tarikfp/hide-bottom-bar-fails-on-ios

https://streamable.com/n5xqej

### Platform

- [ ] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [X] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-bottom-tabs
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [X] @react-navigation/native-stack

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |6.0.6
| @react-navigation/bottom-tabs          |6.1.0
| @react-navigation/native-stack         |6.2.5
| react-native-safe-area-context         |3.3.2
| react-native-screens                   |3.9.0
| react-native-gesture-handler           |1.10.3
| react-native                           |0.66.3

**Comment by tarikpnr** [comment]
> @tarikpnr you're not doing this correctly. You're trying to directly manipulating `style`, while there is a prop for it on `screenOptions`
> 
> I am doing it in an animated way:
> 
> ```tsx
>       tabBarStyle: {
>         transform: [
>           {
>             translateY: transitionAnim,
>           },
>         ],
>       },
> ```
> 
> You could do it like so:
> 
> ```tsx
>       tabBarStyle: {
>         display: 'none'
>       },
> ```
> 
> The key is to use tabBarStyle. But be careful: using `display: none` will be really problematic most of the times. If you really want to use `display: none`, you'll have to set the tabBarStyle `position` to `absolute`. By doing so, you'll end up for the need to compensate for the missing height in every screen using the hook import from `useBottomTabBarHeight` ( `@react-navigation/bottom-tabs`). The easiest way is to use transform.
> 
> All in all, there is no issue with `react-native-screens` regarding the tab bar.

Well, it was just an example. I do know that there is a prop for tabBarStyle in screenOptions, it corresponds to same thing when you use **_custom bottom tab bar._** by giving your **_customTabBar_** component to `tabBar` props of navigator.` display:absolute` is another messy solution unfortunately.

However, I can say that it is resolved with using `enableScreen(false)` in the `useEffect` on `App.js`.
So not sure if its really not because of `react-native-screens` :)

**Comment by hirbod** [comment]
@tarikpnr oh ok, I could only make assumptions based on your ticket. Maybe it is related though, but I was able to use it with enableScreens (but I am also using the native-stack, not sure if you're doing it as well)

**Comment by tarikpnr** [comment]
@hirbod Yes, I use native-stack as well.[ Some says that it might also be related to react-native version. In expo snack you wont see this error.](https://github.com/react-navigation/react-navigation/issues/10432#issuecomment-1072477247)

I really would like to have a prop for `tabBarVisible` as it used to be before... But apparently, it seems not possible from what I can see.

**Comment by wibb36** [comment]
exactly, a prop more like `tabBarHide` is a must. does anyone know how tabBarHideOnKeyboard works?

**Comment by genesisbell** [comment]
any updates on this? I am having this issue

**Comment by sovanyuthchap** [comment]
I am having this issue too. How to fix it?

**Comment by anckaertv** [comment]
Having the same issue. Any fix ?

**Comment by rrsalian16** [comment]
I'm also having the same issue, Any fix?

**Comment by irekrog** [comment]
```ts
sceneContainerStyle={{
  backgroundColor: 'transparent',
}}
```

worked for me

**Comment by edsztukowski** [comment]
I had fixed this issue previously on react native 0.70.1 by setting `enableScreens(false)` but it seems after upgrading to react native 0.72.2, I'm experiencing it again. Any ideas on how to fix this?

**Comment by caglardurmus** [comment]
Any update on this? enableScreens(false) not working on new react native versions(0.73.6).

**Comment by pylvr** [comment]
any  updates on this ?
